assembly/collections/persistentMap.ts

```diff
@@ -170,19 +206,26 @@ export class PersistentMap<K, V> {
    * @param {V} value The new value of the element.
    */
   set(key: K, value: V): void {
+    // assert map size wont overflow
+    assert(this._size < Usize.MAX_VALUE);
```

Comment:
this library should not panic, because there is not try/catch in assemblyscript. So rather return an "error message", we chose to implement the `Result` class in `as-types` package, you can have a look here https://github.com/massalabs/as/blob/main/packages/as-types/assembly/result.ts

it's ok for this PR to leave it like this, but we should at least add a todo comment for using `Result` and/or specify in the jsdoc that this function may panic.